### ISSUE TITLE
[Question]: broadband adapter and SMB load games

### ISSUE THREAD

**Issue by Jerome674** [emukidid/swiss-gc#726]
### Describe your question

Hi !
I use Swiss 0.6 r1362 with LAN Network adaptor and a Synology NAS.
I could read the contents of the directories in FTP and SMB with my GameCube, I have configured both.
I am in SMB 1.0 CIFS
Error message is : device does not support disc images (FTP and SMB).
I can't load any game via SMB, is it implemented?
How can I load games with LAN Network adaptor and a Synology NAS ?
Thanks

**Comment by Extrems** [comment]
Due to resource constraints, only the File Service Protocol is supported.
http://fsp.sourceforge.net/

**Comment by Jerome674** [comment]
What is the use of SMB and FTP in Swiss if they cannot be used?
Just read the files without executing them?
Can we copy data from the network to the SD card and vice versa?

For FSP, it's a mix of FTP and UDP but apart from using port 21, how can it be configured on a Synology PC or NAS?
A part from the page in question, I haven't found anything clear about this FSP protocol.
Could you explain what I could configure to make a FSP connection ?
Thanks

**Comment by Extrems** [comment]
You can load homebrew executable files, play MP3 music, or copy/move files.

I can help with TrueNAS CORE (FreeBSD), Linux and MSYS2 (Windows). I cannot help with Synology.

**Comment by Jerome674** [comment]
You could copy/move files with SMB or FTP or only FSP ?
How do you do this, i don't see extension menu with smb and ftp explorer with Swiss ?

About FSP help :
TrueNAS Core : you add a service/port/protocol or virtualisation ? (if virtualisation, do you have a VM ?)
Windows : you execute an exe and open port in Windows firewall ?

I read that exist FSP server that could start in python, but it's a version without update since 2004.
What is stable to date with Swiss ?
Do you have tutorials with these multiple FSP configuration (Linux and Windows) ?

Thanks !

**Comment by Extrems** [comment]
You enable file management then press the Z Button on a file.

TrueNAS CORE: You make a jail with bridged networking and install the fspd package.
Linux or MSYS2: You build fspd from source.

**Comment by Jerome674** [comment]
What source package can I use with windows ?
I just found a compilation make in 2005.03

With Linux, what are the prerequisites (version to install or compile and files) ?
thanks

**Comment by Extrems** [comment]
Refer to the `INSTALL` file for installation instructions.

**Comment by Jerome674** [comment]
Hello,
I made several tests with the FSP, notably with this script made in python : https://github.com/GoobyCorp/fspd
->Test done on 2 PC
Some games have a little latency when loading some videos and audios but the rendering is very good!

I also did the test from the Synology NAS after installing Python3 on it, then running the script in SSH on it.
I have a better idea of the speed and I see that it often stays around 100KB/s, which is very low.
I also have problems loading videos or even stopping the game.

When copying files from the NAS to the SD2SP2 card, I got these data rates :
-SMB (450KB/s)
-FTP (130KB/s)

I couldn't find any information about the speed of the serial port 1.
Is it limited as low as that?
We are far from the 10/100MB speeds :(
